uniffi_bindgen/src/templates/macros.rs

```diff
@@ -78,34 +78,25 @@
 {% endmatch %}
 {% endmacro %}
 
-{% macro get_arc(obj) -%}
-    {% if obj.threadsafe() %}
-    let _arc = unsafe { std::sync::Arc::from_raw(ptr as *const {{ obj.name() }}) };
-    {% else %}
-    let _arc = unsafe { std::sync::Arc::from_raw(ptr as *const std::sync::Mutex<{{ obj.name() }}>) };
-    {% endif %}
-    // This arc now "owns" the reference but we need an outstanding reference still.
-    std::sync::Arc::into_raw(std::sync::Arc::clone(&_arc));
-{% endmacro %}
-
-
 {% macro to_rs_method_call(obj, meth) -%}
+{% let receiver = meth.first_argument().name().to_string() %}
+{% let receiver_type =  meth.first_argument().type_() %}
 {% match meth.throws() -%}
 {% when Some with (e) -%}
     uniffi::deps::ffi_support::call_with_result(
         err,
         || -> Result<_, {{ e }}> {
-            {% call get_arc(obj) %}
-            let _retval = {{ obj.name() }}::{%- call to_rs_call_with_prefix("_arc", meth, obj) -%}?;
+            let {{ receiver }} = {{ receiver|lift_rs(receiver_type) }}
```

Comment:
So this is actually a little awkward to do in this specific, case but it generalizes better - in a future world where functions are accepting object instances as arguments, the generated code would also be `{{ arg.name()|lift_rs(arg.type_()) }}` and would get all the Arc stuff we've done here for free.